gateway/src/main/kotlin/Event.kt

```diff
@@ -31,7 +28,10 @@ private object NullDecoder : DeserializationStrategy<Nothing?> {
 
     @OptIn(ExperimentalSerializationApi::class)
     override fun deserialize(decoder: Decoder): Nothing? {
-        return decoder.decodeNull()
+        // decodeNull() doesn't consume the literal null therefore parsing doesn't end and in e.g. a heartbeat event
+        // the null gets parsed as a key
+        decoder.decodeNotNullMark()
+        return null
```

Comment:
We should still call `decoder.decodeNull()` for the sake of correctness, even if does literally nothing.
```suggestion
        return decoder.decodeNull()
```

Reply:
Doing nothing sounds fun I like it